code/modules/mob/living/carbon/human/dummy.dm

```diff
@@ -0,0 +1,44 @@
+/mob/living/carbon/human/dummy
+	real_name = "Test Dummy"
+	status_flags = GODMODE|CANPUSH
+	var/in_use = FALSE
+
+//INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy)
+
+/mob/living/carbon/human/dummy/Destroy()
+	in_use = FALSE
+	return ..()
+
+/mob/living/carbon/human/dummy/Life()
+	return
+
+/mob/living/carbon/human/dummy/proc/wipe_state()
+	QDEL_LIST(contents)
+	overlays.Cut()
+
+//Inefficient pooling/caching way.
+GLOBAL_LIST_EMPTY(human_dummy_list)
+
+/proc/generate_or_wait_for_human_dummy(slotkey)
+	if(!slotkey)
+		return new /mob/living/carbon/human/dummy
+	var/mob/living/carbon/human/dummy/D = GLOB.human_dummy_list[slotkey]
+	if(istype(D))
+		while(!D.in_use)
+			stoplag()
```

Comment:
I think our `stoplag` and TG's `stoplag` are very different. You may want to look into this; this may be part of your problem.

Reply:
yeah i made it this cuase its essientially what TGs UNTIL() define is doing.

Reply:
and as posted to you on discord..yeah..they are diffrent...what do you suggest here?

Reply:
`CHECK_TICK`? :thonkong:

Reply:
it also uses stoplag..not sure if it applys here. i may need someone to give me another opinion. I DID think about that.

Reply:
okay yeah after testing its stopping at var/mob/living/carbon/human/dummy/D = GLOB.human_dummy_list[slotkey] running an istype(D)print returns 0. and it dosen't get to the second print, so its hitting the return.